### PR TITLE
Feature/eint 27 update readme rokt rokt demo ios

### PR DESCRIPTION
**Pull request by danial-rokt**
### Background ###

Readme updated to transfer the knowledge 

### Checklist: ###

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have verified that CI completes successfully.
- [ ] All insignificant commits have been squashed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,66 @@
...
-Rokt iOS Demo Application
+Rokt Demo application is a sample app built to showcase Rokt SDK functionality.
+
+## Requirements
+
+XCode 13 and above. Project is configured to run on iOS 13.0 and above.
+
+## Project architecture
+
+This project it is implemented based on MVVM pattern with SwiftUI.
```
Comment (DJingSeo) [inline README.md:10]: description for MVVM needed

File: README.md [hunk 2]
```diff
@@ -1,2 +1,66 @@
...
+It contains UI, Model, Services, Utils and Resources.
```
Comment (DJingSeo) [inline README.md:11]: Explains each

File: README.md [hunk 3]
```diff
@@ -1,2 +1,66 @@
...
+![UI diagram](diagram.png)
+
+## Where to start
+
+As it is defined on SceneDelegate, the starting point of the application is HomePageView located in `UI/Home` folder.
+
+## How to release the app
+- Update `Version` and `Build` in RoktDemo target
+- Commit & push and merge it to the release branch
+- Aprove the **hold** on **CircleCI** and wait till it finished
```
Comment (DJingSeo) [inline README.md:26]: link to circleCI

File: README.md [hunk 4]
```diff
@@ -1,2 +1,66 @@
...
+All the other resources of the application is located in Resources. 
+![UI diagram](diagram.png)
+
+## Where to start
+
+As it is defined on SceneDelegate, the starting point of the application is HomePageView located in `UI/Home` folder.
+
+## How to release the app
+- Update `Version` and `Build` in RoktDemo target
+- Commit & push and merge it to the release branch
```
Comment (DJingSeo) [inline README.md:25]: what's release brance?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,66 @@
...
+- Aprove the **hold** on **CircleCI** and wait till it finished
+- Follow the instructions on itunesconnect
+
+## FAQ
+
+### How to setup the environment?
+- run `Pod install`
```
Comment (DJingSeo) [inline README.md:32]: where to execute?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,66 @@
...
+* **version 1.1.x** - The Brand demos added
+
+### How to run tests locally?
+
+Select the RoktDemo scheme and press command + U, or Product -> Test.
+Tests can be found in `RoktDemoTests` and `RoktDemoUIITests`.
+
+### How to run tests on CircleCI?
+
+Just push a change to any of the branches and tests will be run.
```
Comment (DJingSeo) [inline README.md:46]: where to check out the outcome

File: README.md [hunk 7]
```diff
@@ -1,2 +1,66 @@
...
+
+### Something is not working, what do I do?
+
+In case you run into any problems you can try:
+
+* Clean project: Product -> Clean
```
Comment (DJingSeo) [inline README.md:52]: What clean means?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,66 @@
...
+
+### Where are dependencies defined?
+Cocoapods used as dependency manager of this repository. All project dependencies are defined on the podfile
```
Comment (DJingSeo) [inline README.md:55]: What's cocoapod?

**Comment by DJingSeo** [comment]
Take a read the Documentation section of https://docs.google.com/document/d/1xfgyu7qF_TS6_V1z1ndkgrxcZPUOL8l57lSUvjwTV7Y/edit#heading=h.dej4me02wx5 and reflect it in this README

File: README.md (hunk 6 above)
Comment (danial-rokt) [reply README.md:46]: I don't think it is a good idea to put the link to CircleCI build as this is a public repository

File: README.md (hunk 2 above)
Comment (danial-rokt) [reply README.md:11]: A short description of each layer is introduced in the next paragraph.

File: README.md (hunk 3 above)
Comment (danial-rokt) [reply README.md:26]: Thanks for mentioning, I would put the link to CircleCI but not the actual build

File: README.md (hunk 4 above)
Comment (danial-rokt) [reply README.md:25]: will add a branching strategy here

Comment (danial-rokt) [reply README.md:31]: added

File: README.md [hunk 9]
```diff
@@ -1,2 +1,81 @@
...
-Rokt iOS Demo Application
+Rokt Demo application is a sample app built to showcase Rokt SDK functionality.
+
+## Requirements
+
+XCode 13 and above. Project is configured to run on iOS 13.0 and above.
+
+## Project architecture
+
+This project it is implemented based on MVVM pattern with SwiftUI.
```
Comment (sahil-rokt) [inline README.md:10]: remove "it". or This project is implemented with MVVM pattern using SwiftUI.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,81 @@
...
+## Project architecture
+
+This project it is implemented based on MVVM pattern with SwiftUI. 
+MVVM stands for Model-View-ViewModel, an architecture pattern that is structured to separate program logic and user interface controls. It has Model, View and ViewModel separation. 
+
+This project contains UI, Model, Services, Utils and Resources groups.
+
+UI: View and ViewModels are located in the UI Group.
+
+Services & Model: Services represent the network logics of the application and model represnts the data.
```
Comment (sahil-rokt) [inline README.md:17]: typo in represnts

File: README.md [hunk 11]
```diff
@@ -1,2 +1,81 @@
...
+### What are the versions?
+
+* **release 1.x** - The first version of Demo application
+* **release 1.1.x** - The PreDefined brand demos added
+
+### What are the branches?
+
+here are main branches coresponding to each version : **Master**, **Release branches** and **Features branches**
+
+* **master** - Master branch which corrently is not used
```
Comment (sahil-rokt) [inline README.md:49]: typo corrently

File: README.md [hunk 12]
```diff
@@ -1,2 +1,81 @@
...
+* **feature branches** - After every push to this branch swift lint and tests are run to ensure no breaking changes are allowed.
+
+### How to run tests locally?
+
+Select the RoktDemo scheme and press command + U, or Product -> Test.
+Tests can be found in `RoktDemoTests` and `RoktDemoUIITests`.
+
+### How to run tests on CircleCI?
+
+Just push a change to any of the branches and tests will be run. The result of the test would be visible in the [CircleCi](https://circleci.com/).
```
Comment (sahil-rokt) [inline README.md:60]: are we not giving the correct circleCI path ?

Comment (danial-rokt) [reply README.md:60]: That is what I am not sure yet. 
This is an open-source project. Do you think putting the full path there is a good idea?

Comment (sahil-rokt) [reply README.md:60]: yes most open-source-projects have a link to CI/CD pipeline. There should not be any issue.

File: README.md (hunk 10 above)
Comment (danial-rokt) [reply README.md:17]: Good one

File: README.md (hunk 9 above)
Comment (danial-rokt) [reply README.md:10]: Done

File: README.md (hunk 8 above)
Comment (danial-rokt) [reply README.md]: Updated and description of Cocoapds added.

File: README.md (hunk 7 above)
Comment (danial-rokt) [reply README.md]: for iOS projects, cleaning the project is common knowledge, but as you mentioned, I added more description here.

File: README.md [hunk 13]
```diff
@@ -1,2 +1,84 @@
...
+## How to release the app
+- Update `Version` and `Build` in RoktDemo target
+- Commit & push and merge it to the release branch
+- Aprove the **hold** on **[CircleCI](https://circleci.com/)** and wait till it finished
+- Follow the instructions on itunesconnect
+
+## FAQ
+
+### How to setup the environment?
+- run `Pod install` on the main folder of iOS application where podfile exists.
```
Comment (DJingSeo) [inline README.md:41]: What pods will be installed?
How to test beta version or non-released iOS SDK with this demo app?

File: README.md [hunk 14]
```diff
@@ -1,2 +1,84 @@
...
+
+### What are the versions?
+
+* **release 1.x** - The first version of Demo application
+* **release 1.1.x** - The PreDefined brand demos added
```
Comment (DJingSeo) [inline README.md:46]: What's the versioning rule?

File: README.md (hunk 12 above)
Comment (DJingSeo) [reply README.md:63]: We should have the link

File: README.md [hunk 15]
```diff
@@ -1,2 +1,84 @@
...
+
+### Something is not working, what do I do?
+
+In case you run into any problems you can try:
+
+* Clean the build folder of the project. Select menu: Product -> Clean Build Folder
+
+### Where are dependencies defined?
+
+All project dependencies are defined on the podfile. Cocoapods used as dependency manager of this repository. The full description of Cocoapods could be find here [here](https://cocoapods.org/)
```
Comment (DJingSeo) [inline README.md:73]: Better to list what major dependencies this app has.

Comment (DJingSeo) [reply README.md:73]: Short summary of descrption of Cocoapods needed

File: README.md [hunk 16]
```diff
@@ -1,2 +1,84 @@
...
+![UI diagram](diagram.png)
+
+## Where to start
+
+As it is defined on SceneDelegate, the starting point of the application is HomePageView located in `UI/Home` folder.
+
+## How to release the app
+- Update `Version` and `Build` in RoktDemo target
+- Commit & push and merge it to the release branch
+- Aprove the **hold** on **[CircleCI](https://circleci.com/)** and wait till it finished
```
Comment (DJingSeo) [inline README.md:35]: How to get this permission?

File: README.md [hunk 17]
```diff
@@ -1,2 +1,84 @@
 # rokt-demo-ios
-Rokt iOS Demo Application
+Rokt Demo application is a sample app built to showcase Rokt SDK functionality.
```
Comment (DJingSeo) [inline README.md:2]: Who use? 

To whom this app distributes to?

File: README.md [hunk 18]
```diff
@@ -1,2 +1,84 @@
...
+## Resident Experts
+- Danial Motahari - danial.motahari@rokt.com
+
+## Requirements
+
+XCode 13 and above. Project is configured to run on iOS 13.0 and above.
+
+## Project architecture
+
+This project is implemented based on MVVM pattern with SwiftUI.
```
Comment (DJingSeo) [inline README.md:13]: Should explain SwiftUI even briefly

File: README.md [hunk 19]
```diff
@@ -1,2 +1,84 @@
...
+
+Services & Model: Services represent the network logics of the application and model represents the data.
+
+Utils: are shared between Groups of the application.
+
+Resources: All the other resources of the application is located in Resources. 
+
+![UI diagram](diagram.png)
+
+## Where to start
```
Comment (DJingSeo) [inline README.md:28]: start... looking? code? I am not sure what is the purpose of this section

File: README.md [hunk 20]
```diff
@@ -1,2 +1,84 @@
...
+
+## How to release the app
+- Update `Version` and `Build` in RoktDemo target
+- Commit & push and merge it to the release branch
+- Aprove the **hold** on **[CircleCI](https://circleci.com/)** and wait till it finished
+- Follow the instructions on itunesconnect
+
+## FAQ
+
+### How to setup the environment?
```
Comment (DJingSeo) [inline README.md:40]: This shouldn't be a part of FAQ but another major section

File: README.md [hunk 21]
```diff
@@ -1,2 +1,84 @@
...
+
+here are main branches coresponding to each version : **Master**, **Release branches** and **Features branches**
+
+* **master** - Master branch which currently is not used
+* **release branches** - Prod environment. After every push to this branch tests are run to ensure no breaking changes are allowed and after approval. Releasing to TestFlight could be enabled after all the tests passes. 
+* **feature branches** - After every push to this branch swift lint and tests are run to ensure no breaking changes are allowed.
+
+### How to run tests locally?
+
+Select the RoktDemo scheme and press command + U, or Product -> Test.
```
Comment (DJingSeo) [inline README.md:58]: On Xcode?

File: README.md (hunk 6 above)
Comment (DJingSeo) [reply README.md]: Why not? CircleCI has its own authentication

**Comment by DJingSeo** [comment]
let's assume that the expected readers for this README are not us but grad engineers in NYC. They should be able to understand what this repo is for, able to debug on local, fix a small bug, test and deploy it to prod.

File: README.md (hunk 19 above)
Comment (danial-rokt) [reply README.md:28]: I thought it is a good practice to point to a starting point of the application. As we are not using a VIPER design pattern that has a specific layer for routing, `where to start` is a good point specially for grad engineers in NYC. 
However, I would remove it.

File: README.md (hunk 16 above)
Comment (danial-rokt) [reply README.md:35]: Sure I would include this one, then I would make sure every other Readme update has this information as well.

File: README.md (hunk 15 above)
Comment (danial-rokt) [reply README.md:73]: The official description of Cocoapds on their official website https://cocoapods.org/ is 3 lines which two lines of them are just showing off :D.
Sure I would add more description.

File: README.md (hunk 17 above)
Comment (danial-rokt) [reply README.md:2]: > let's assume that the expected readers for this README are not us but grad engineers in NYC. They should be able to understand what this repo is for, able to debug on local, fix a small bug, test and deploy it to prod.

I think the user of this repository, should at least have basic iOS knowledge if they want to test and deploy it to prod. The professional level of this readme would be reduced a lot as we are explaining every single terminology like MVVM. But I would reflect all the changes you mentioned here.

File: README.md (hunk 12 above)
Comment (danial-rokt) [reply README.md]: Done

File: README.md (hunk 13 above)
Comment (danial-rokt) [reply README.md]: It added separately

File: README.md (hunk 14 above)
Comment (danial-rokt) [reply README.md]: Done

File: README.md (hunk 17 above)
Comment (danial-rokt) [reply README.md]: Updated entirely

File: README.md (hunk 18 above)
Comment (danial-rokt) [reply README.md:23]: SwiftUI explanation added

File: README.md (hunk 20 above)
Comment (danial-rokt) [reply README.md]: done

File: README.md (hunk 21 above)
Comment (danial-rokt) [reply README.md]: done

**Review by DJingSeo — CHANGES_REQUESTED** [review]
README looks much better than before.
This repo has a security vulnerability that should be addressed.

**Comment by danial-rokt** [comment]
> README looks much better than before. This repo has a security vulnerability that should be addressed.

the security vulnerability is addressed in this PR. Thanks for mentioning it.

File: Gemfile.lock [hunk 22]
```diff
@@ -142,7 +142,7 @@ GEM
     http-cookie (1.0.4)
       domain_name (~> 0.5)
     httpclient (2.8.3)
-    jmespath (1.4.0)
+    jmespath (1.6.1)
```
Comment (DJingSeo) [inline Gemfile.lock:145]: +1

**Review by DJingSeo — APPROVED** [review]

**Pull request merged**